docker/Dockerfile

```diff
@@ -1,80 +1,106 @@
-RUN rm -rf /TileDB-0.6.1
+RUN cd / && wget https://github.com/TileDB-Inc/TileDB/archive/1.3.1.tar.gz && \
+    tar xf 1.3.1.tar.gz && cd TileDB-1.3.1 && mkdir build && cd build && \
+    ../bootstrap --prefix=/usr/local/ && make -j && make install-tiledb
+RUN rm -rf /TileDB-1.3.1
 
-# Valijson
-RUN cd / && git clone https://github.com/tristanpenman/valijson.git
-RUN cd valijson && cp -r include/* /usr/local/include/
+# Faiss
+RUN cd / && \
+    wget https://github.com/facebookresearch/faiss/archive/v1.4.0.tar.gz && \
+    tar xf v1.4.0.tar.gz && \
+    cd faiss-1.4.0 && rm CMakeLists.txt && \
+    wget https://gist.githubusercontent.com/luisremis/758c71c6a86cb3e4fe2311c415e07547/raw/3c51b5ed7d3f16a4d684d9ea524602c9f1f8593a/CMakeLists.txt && \
+    mkdir build && cd build && cmake ../ && make -j
+RUN mkdir /usr/local/include/faiss/
+RUN cp -r faiss-1.4.0/* /usr/local/include/faiss/
+RUN mv faiss-1.4.0/build/lib/libfaiss.so /usr/local/lib/
+RUN rm -rf faiss-1.4*
 
-# OpenCV
-RUN mkdir /opencv && cd /opencv && wget -O opencv.zip https://github.com/opencv/opencv/archive/3.3.0.zip
-RUN cd /opencv && unzip opencv.zip && cd opencv-3.3.0 && mkdir build && cd build && cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_INSTALL_PREFIX=/usr/local .. && make -j $(cat /proc/cpuinfo | wc -l) && make install && rm -rf /opencv
+# PMGD install
+RUN cd / && wget https://github.com/intellabs/pmgd/archive/v2.0.0.tar.gz && \
+    tar xf v2.0.0.tar.gz && mv pmgd-2.0.0 pmgd && cd pmgd && \
+    make PMOPT=MSYNC
 
 # VCL install
-RUN cd / && wget https://github.com/intellabs/vcl/archive/v0.1.0.tar.gz && tar xf v0.1.0.tar.gz
-RUN mv vcl-0.1.0 vcl && cd vcl && scons -j16
+RUN cd / && wget https://github.com/intellabs/vcl/archive/v1.0.0.tar.gz && \
+    tar xf v1.0.0.tar.gz && mv vcl-1.0.0 vcl && cd vcl && scons -j16
 
-# PMGD install
-RUN cd / && wget https://github.com/intellabs/pmgd/archive/v1.0.0.tar.gz && tar xf v1.0.0.tar.gz
-RUN mv pmgd-1.0.0 pmgd && cd pmgd && make
-RUN rm -rf /*.tar.gz
+# Valijson
+RUN cd / && git clone https://github.com/tristanpenman/valijson.git && \
+    cd valijson && cp -r include/* /usr/local/include/
 
 # VDMS install
-RUN cd / && wget https://github.com/intellabs/vdms/archive/v1.0.0.tar.gz && tar xf v1.0.0.tar.gz
-RUN mv vdms-1.0.0 vdms && cd vdms && mkdir db && scons -j16 INTEL_PATH=/
+RUN git clone https://github.com/intellabs/vdms/ && \
+    cd vdms && git checkout develop && \
+    mkdir db && scons -j16 INTEL_PATH=/
 
 RUN rm -rf /*.tar.gz
 RUN rm -rf /var/lib/apt/lists/* /root/.cache
 
+# Setup entry point
+
 RUN echo '#!/bin/bash' > /start.sh
-RUN echo 'export LD_LIBRARY_PATH=/pmgd/lib:$(find /usr/local/lib/ / -type f -name "*.so" | xargs dirname | sort | uniq | tr "\n" ":")' >> /start.sh
+RUN echo 'export LD_LIBRARY_PATH=/vdms/utils:/pmgd/lib:$(find /usr/local/lib/ / -type f -name "*.so" | xargs dirname | sort | uniq | tr "\n" ":")' >> /start.sh
 RUN echo 'cd /vdms' >> /start.sh
 RUN echo 'rm -rf db' >> /start.sh
```

Comment:
CAn we add a line here to mkgraph using reasonable params like multiple allocators? Sadly we cannot create one with indexes and such but something. Maybe in the docker with samples

Reply:
this will use the default number of allocators that can be set through vdms config file (we could change our default). Also, users can modify this param and re-run vdms in the instance, if needed.